### PR TITLE
add sl micro 6.0 extras repo for rpm resolution

### PR DESCRIPTION
**Pull request by jdob**
closes: #591 

See #591 for more information. The summary is that EIB 1.2 only needs to support SL Micro 6.0. This patch is a temporary fix that relies on that assumption, hardcoding the version into the repo URL.

Eventually, this will need to be migrated to dynamically introduce the SL Micro version based on the base image. I know how to get that, from `/etc/os-releases`. For the RAW image case, it's somewhat simple: a short guestfish script to extract the file and then some basic parsing for a variable. For ISOs, however, it's more invasive, so we currently unsquash the raw image at rebuild time, meaning it's not available early enough to extract the Micro version. It's fixable, but much more invasive and out of scope for 1.2.

To verify this, try to install the package `cpp`. In SL Micro 6.x, this is no longer in the base repository, now located in the extras repo that is added by this patch.

### CODE REVIEW

**Comment by e-minguez** [comment]
I don't think we want to enable this on every build. Honestly I was thinking of enabling a new parameter under the packages section like the repos one but for 'extraProducts'. AFAIK there are at least two, the Extras and the one for SUMA (I don't remember the name exactly).
So users will need to put the list of extraProducts they need to enable and then pass those directly to suseconnect -p

**Comment by jdob** [comment]
There's only one as far as I can see, the extras. That's one reason why I wanted to opt for a simplistic approach for the time being; I'm not sure it's clear how those will shake out, which may lead us to want a more fine grained ability to indicate which to enable. 

Similar to what I said in #591, I'd rather not introduce new definition parameters until we have a better idea of what the end goal is from Micro side. Once an API change like that is out in the wild, we need to continue to support it and deal with a deprecation process, neither of which are particularly fun. IMO, definition additions like this require more information than we have right now to do it correctly.

For example, adding `extraProducts` as a list is excessive if there is only one extra repository, whereas making that a boolean is potentially limiting if they do introduce more than one extra repository. Once their strategy on using these repos becomes more clear, we can make this more configurable.

Let me flip the question around. You said you don't think we want to enable this on every build. Why not? If there's a compelling reason, we may have to revisit this approach, but as of right now I think the tradeoff of enabling it in every build for a less volatile API later is worth it.

**Comment by e-minguez** [comment]
For 5.5 there are at least two, package hub and live patching https://documentation.suse.com/sle-micro/5.5/single-html/SLE-Micro-deployment/#sec-adding-extension so I guess these changes won't work on 5.5
For 6.X I cannot find it right now

**Review by dbw7 — DISMISSED** [review]

**Review by atanasdinov — APPROVED** [review]

**Comment by jdob** [comment]
I converted this to draft for now. I'm playing around with adding a definition flag to enable it.

**Comment by jdob** [comment]
Ok, so I tested this and it works. The build will fail if `cpp` is a specified package and the `enableExtras` flag isn't included.

I went with a single flag for the implementation. From the sound of it, I don't anticipate they are going to introduce a ton of these sorts of repos which would necessitate a list. If we do need to deprecate this in the future, it shouldn't be too difficult.

I explicitly did not add any of the validation pieces around the api version in this PR. Once we figure out what's going on with dual stack support, we can add that in a separate PR follow up.

@atanasdinov @dbw7 - What do you guys think?

File: pkg/rpm/resolver/templates/rpm-resolution.sh.tpl [hunk 1]
```diff
@@ -10,9 +10,13 @@ set -euo pipefail
 #  LocalGPGList - list of local GPG keys that will be imported in the resolver image
 #  NoGPGCheck   - when set to true skips the GPG validation for all third-party repositories and local RPMs
 #  Arch         - sets the architecture of the rpm packages to pull
+#  EnableExtras - registers the SL-Micro-Extras repo for use in resolution
 
 {{ if ne .RegCode "" }}
 suseconnect -r {{ .RegCode }}
+{{ if $.EnableExtras -}}
```
Comment (atanasdinov) [inline pkg/rpm/resolver/templates/rpm-resolution.sh.tpl:17]: Nit: The dollar sign shouldn't be necessary.

**Review by atanasdinov — APPROVED** [review]
LGTM!

File: pkg/rpm/resolver/templates/rpm-resolution.sh.tpl (hunk 1 above)
Comment (jdob) [reply pkg/rpm/resolver/templates/rpm-resolution.sh.tpl:17]: Admittedly, I didn't even pull up the template syntax docs. I looked at the GPG conditional a few lines below and followed that model. Still, I appreciate the heads up.

**Review by atanasdinov — APPROVED** [review]

**Pull request merged**